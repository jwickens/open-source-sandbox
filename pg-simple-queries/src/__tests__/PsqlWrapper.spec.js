/**
 * @flow
 */

const { Db, PsqlWrapper } = require('../')
const path = require('path')
const sqlDirectory = path.join(__dirname, 'sql')
const randomNum = Math.floor(Math.random() * 1000)

let db1

beforeAll(async () => {
  db1 = new Db({sqlDirectory, overrideEnv: true, schemaBase: 'test-psql-wrapper' + randomNum, schemaVersion: 0})
  await db1.sync()
})

afterAll(async () => {
  await db1.close()
})

describe('psql wrapper', () => {
  it('can initialize', async () => {
    const psql = new PsqlWrapper(db1)
    await psql.isInitialized
  }, 10000)

  it('can describe a table', async () => {
    const psql = new PsqlWrapper(db1)
    const desc = await psql.describeTable('test')
    expect(desc).toMatchSnapshot()
  })
})
