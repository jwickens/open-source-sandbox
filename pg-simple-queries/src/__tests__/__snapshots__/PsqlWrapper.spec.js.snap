// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`psql wrapper can describe a table 1`] = `
"Table \\"<schema_name>.test\\"
Column|Type|Modifiers
id|integer|not null default nextval('test_id_seq'::regclass)
test_name|text|
message|text|
tags|text[]|
Indexes:
\\"test_pkey\\" PRIMARY KEY, btree (id)
\\"trgm_idx\\" gin (message public.gin_trgm_ops)"
`;
